### PR TITLE
chore(configurations): add opa helm chart

### PR DESCRIPTION
**Pull request by AnguIar**
| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Chore            | ✔                                                                       |

Add opa helm chart to opa-la

### CODE REVIEW

File: helm/templates/configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{{- $chartName := include "opa.name" . -}}
+{{- if .Values.enabled -}}
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Release.Name }}-{{ $chartName }}-configmap
+data:
+  OPA_CONFIG: {{ tpl (.Files.Get "config/opa.yaml") . | quote }}
```
Comment (CptSchnitz) [inline helm/templates/configmap.yaml:8]: make path configurable with a default value

File: helm/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+            mountPath: /opa/opa.yaml
+            readOnly: true
+            subPath: opa.yaml
+          args:
+          - "run"
+          - "--server"
+          - "--addr=0.0.0.0:8181"
+          - "--diagnostic-addr=0.0.0.0:8282"
+          - "--set=plugins.envoy_ext_authz_grpc.add=:{{ .Values.image.tag }}"
+          - "--set=plugins.envoy_ext_authz_grpc.path=envoy/authz/allow"
```
Comment (CptSchnitz) [inline helm/templates/deployment.yaml:64]: i dont think we need grpc.

File: helm/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+          {{- if .Values.enableLogging }}
+          - "--set=decision_logs.console=true"
```
Comment (CptSchnitz) [inline helm/templates/deployment.yaml:66]: configure this using env or the config file

File: helm/values.yaml [hunk 4]
```diff
@@ -28,6 +28,28 @@ dbConfig: &dbConfig
...
+
+resources:
+  enabled: true
+  value:
+    limits:
+      cpu: 100m
+      memory: 128Mi
+    requests:
+      cpu: 100m
+      memory: 128Mi
```
Comment (CptSchnitz) [inline helm/values.yaml:51]: it would be nice to have the values ready for outnet (4 * request = limit)

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

File: helm/values.yaml (hunk 4 above)
Comment (AnguIar) [reply helm/values.yaml:51]: it doesn't really matter as we need to make sizing for opa

File: helm/templates/configmap.yaml (hunk 1 above)
Comment (AnguIar) [reply helm/templates/configmap.yaml]: why do wee need it as a configuration?

**Review by CptSchnitz — APPROVED** [review]

File: helm/charts/opa/Chart.yaml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v2
+name: opa
+description: A Helm chart for opa
+type: application
+version: 0.0.0
+appVersion: 0.0.0
```
Comment (syncush) [inline helm/charts/opa/Chart.yaml:6]: lets setup a base version of v1.0.0

File: helm/charts/opa/templates/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+            mountPath: /opa/opa.yaml
+            readOnly: true
+            subPath: opa.yaml
+          args:
+          - "run"
+          - "--server"
+          - "--addr=0.0.0.0:8181"
+          - "--diagnostic-addr=0.0.0.0:8282"
+          - "--config-file"
+          - "/opa/opa.yaml"
```
Comment (syncush) [inline helm/charts/opa/templates/deployment.yaml:64]: should we extract the args to values.yaml ?

**Review by syncush — CHANGES_REQUESTED** [review]
What about the monitoring of OPA ?

https://www.openpolicyagent.org/docs/latest/monitoring/

we should support it

Does auth-manager have telemetry configurations available? is it under the globals?

**Review by syncush — APPROVED** [review]
what about opa metrics?

File: helm/charts/opa/templates/ingress.yaml [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    nginx.org/location-snippets: |
+      if ($request_method = OPTIONS) {
+        return 204;
+      }
+    {{- if .Values.ingress.cors.enabled }}
+      add_header 'Access-Control-Allow-Origin' '{{- .Values.ingress.cors.origin -}}';
+      add_header 'Access-Control-Max-Age' 3600;
+      add_header 'Access-Control-Expose-Headers' 'Content-Length';
+      add_header 'Access-Control-Allow-Headers' '*';
+    {{- end }}
```
Comment (syncush) [inline helm/charts/opa/templates/ingress.yaml:21]: why do you care about cors?

File: helm/charts/opa/templates/route.yaml [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    haproxy.router.openshift.io/rewrite-target: /
+    {{- if .Values.route.timeout.enabled }}
+    haproxy.router.openshift.io/timeout: {{ .Values.route.timeout.duration }}
+    {{- end }}
+spec:
+  path:  {{ .Values.route.path | default "/" }}
+  to:
+    kind: Service
+    name: {{ $serviceName }}
+  port:
```
Comment (syncush) [inline helm/charts/opa/templates/route.yaml:20]: hostname? TLS? path ?

**Review by syncush — CHANGES_REQUESTED** [review]

**Review by syncush — APPROVED** [review]

**Pull request merged**